DirectProgramming/C++SYCL_FPGA/Tutorials/Features/IPAuthoring_interfaces_overview/host-pipes/src/vector_add.cpp

```diff
@@ -0,0 +1,105 @@
+#include <iostream>
+
+// oneAPI headers
+
+#include <sycl/ext/intel/experimental/pipes.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>  // sycl.hpp comes first to prevent compiler crash
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class SimpleVAdd_pipes;
+
+template <typename pipeA, typename pipeB, typename pipeC>
+struct SimpleVAddKernel_pipes {
```

Comment:
```suggestion
struct SimpleVAddKernelPipes {
```

code samples committee doesn't like structs with `_` in them